examples/using-router/components/CustomLink.tsx

```diff
@@ -0,0 +1,47 @@
+import { useRouter } from 'next/router'
+import { useEffect, ReactNode, HTMLAttributes } from 'react'
+
+type CustomLinkProps = {
+  children: ReactNode
+  href: string
+  prefetch?: boolean
+  replace?: boolean
+  shallow?: boolean
+} & HTMLAttributes<HTMLAnchorElement>
+
+// typically you want to use `next/link` for this usecase
+// but this example shows how you can also access the router
```

Comment:
I don't think we need to showcase a way to recreate `next/link`. :thinking:

Reply:
oh actually I've just migrated this comment from here:

https://github.com/vercel/next.js/blob/1264196897e11a0561ff9ca9df10c25fb5ee718e/examples/using-router/components/Header.js#L12-L33

Do you have alternative suggestions to show some examples of this imperative api?

Reply:
I see. In that case, let's leave it then for now. :+1: Thanks for clarifying!